pkg/sql/sem/tree/pretty_test.go

```diff
@@ -152,7 +152,7 @@ func runTestPrettyData(
 				t.Fatalf("expected:\n%s\ngot:\n%s", expect, got)
 			}
 
-			sqlutils.VerifyStatementPrettyRoundtrip(t, string(sql))
+			sqlutils.VerifyStatementPrettyRoundtrip(t, res[len(res)-1])
```

Comment:
Why did this need to change?

Reply:
The pretty-printer now "pulls" all the column definitions to appear before the constraint definitions. This means that the order is not preserved any more -- it does not round trip using the initial SQL, only after the first round of pretty-printing.

Reply:
Hmm then I may need some convincing that the reordering is worth it. The purpose of this test is to make sure that pretty printing doesn't miss anything from the original SQL. If we are now testing the round trip of the post-prettified string, the test is no longer testing what it was designed to.

Here's an idea: instead of reordering all statements in order to align names and types, only align adjacent ColumnTableDefs. This would preserve the user-specified ordering, and probably look good for the vast majority of create tables. I could see users be displeased if they spent time grouping their columns and, say, index definitions in a specific and meaningful way and then sqlfmt changes that ordering.

Reply:
> Here's an idea: instead of reordering all statements in order to align names and types, only align adjacent ColumnTableDefs. This would preserve the user-specified ordering, and probably look good for the vast majority of create tables.

Good idea. Done.